examples/dataset_example/CMakeLists.txt

```
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.24)
+project(hello_world_project VERSION 1.0.0 LANGUAGES CXX)
```

I don't think we want to leave everything as a `hello_world_project`
```suggestion
project(dataset_example_project VERSION 1.0.0 LANGUAGES CXX)
```